python/tvm/micro/build.py

```diff
@@ -162,7 +158,12 @@ def build_static_runtime(
 
         libs.append(compiler.library(lib_build_dir, lib_srcs, lib_opts))
 
-    libs.append(compiler.library(mod_build_dir, [mod_src_path], generated_lib_opts))
+    libs.append(
```

Comment:
No, the export_library will create a temporary directory to save source files and it will not live beyond the scope of the function. So it did not occur to me saving the sources is a feature (I thought it was a side-effect of this flow). Thus, I ll just say what I generally do: use pdb and do get_source to see the c-source created. Anyhow, do we think this would be a user-facing feature ?

if so we can do two things : 
A1 : Traverse module hierarchy and dump the sources here independently of export library, here.
A2 :  More generally  we can introduce an attr to target_kind "c" in the lines of something like "-dbgdir=<dbgdir>" and make the export_library to use that directory the dump the intermediary sources.

I think A2 would be more generic, but would be beyond the scope of this PR. 
Thus if people agree, I could just do A1 here for now with a TODO for A2 at some point. 
What do you think ?

Reply:
I think in the future we should consider generating full projects (I.e. zephyr projects) at which point we should have a defined way to describe module and library sources to the generators. saving generated code to the workspace tree means that (if the Workspace is operating with debug=True, you get source included in your debugger. that can come in handy when the thing causing a CHECK-fail is memory overflow from a TVMBackendAllocWorkspace inside an operator function. So I think we want A1 but we also want to compile the library from the user-accessible source files.

Reply:
In order to satisfy that requirement, we could still make the export_library create the artifacts in a user given workspace directory. Thus, we would not have this split flow of exporting the IRModules in tvm vs utvm and thats what I meant by A2. 

On other hand of generating projects, I think we should discuss it in a RFC. My personal feeling it should be a layer above the TVM stack (similiar tvmc) to collect the artifacts produced out of tvm to generate such projects rather than making tvm a project generator. However, I dont have a strong opinion on either.

Thoughts ? : @tqchen

Reply:
I agree that we should discuss with an RFC. however, I don't want to break support for GDB debugging with source now.

regarding the target_kind attribute--we do need to be a bit careful with those, because they serve as keys to the autotuning logs. I don't know that a debug source directory should be part of identifying an autotuning result.

it seems like perhaps `export_library` could just allow you to specify the tempdir as a kwarg?

Reply:
I added another optional argument for export_library as workspace_dir. Technically, I could've use kwarg but I think that is for the purpose of pass through arguments to fcompile function. Moreover, I think have a workspace dir should be useful for debugging anyway for different use cases of export_library. See if this work :) .

Reply:
https://github.com/apache/tvm/pull/7002/ -- Implemented here.